search/v2/geocoder.cpp

```diff
@@ -689,6 +750,44 @@ void Geocoder::FillLocalitiesTable(MwmContext const & context)
   }
 }
 
+void Geocoder::FillVillageLocalities()
+{
+  vector<Locality> preLocalities;
+  FillLocalityCandidates(m_villages.get(), kMaxNumVillages, preLocalities);
+
+  size_t numVillages = 0;
+
+  for (auto & l : preLocalities)
+  {
+    FeatureType ft;
+    m_context->m_vector.GetByIndex(l.m_featureId, ft);
+
+    switch (m_model.GetSearchType(ft))
+    {
+    case SearchModel::SEARCH_TYPE_VILLAGE:
+    {
+      if (numVillages < kMaxNumVillages && ft.GetFeatureType() == feature::GEOM_POINT)
+      {
+        ++numVillages;
+        City village = l;
+        village.m_rect = MercatorBounds::RectByCenterXYAndSizeInMeters(
+            ft.GetCenter(), ftypes::GetRadiusByPopulation(ft.GetPopulation()));
```

Comment:
OK, for consistency with cities leave it as is, but note that ft.GetCenter() works only for points and may work incorrectly with complex shapes.